### DISCUSSION TITLE
helm upgrade 9.2 to 9.4 failed

### DISCUSSION THREAD

**Discussion by maristeslk** [apache/skywalking#10625] [Q&A]
NAME                    CHART VERSION   APP VERSION     DESCRIPTION                 
skywalking/skywalking   4.3.0            Apache SkyWalking APM System

"I tried two methods to upgrade to version 9.4.
methods 1
 I used the command 'helm upgrade skywalking skywalking/skywalking -n skywalking --set oap.image.tag=9.4.0 --set ui.image.tag=9.4.0 --set oap.storageType=elasticsearch', but I received an error message which states 'Error: UPGRADE FAILED: cannot patch "elasticsearch-master" with kind StatefulSet: StatefulSet.apps "elasticsearch-master" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden && cannot patch "skywalking-es-init" with kind Job: Job.batch "skywalking-es-init" is invalid: spec.template: Invalid value: core.PodTemplateSpec...'. 
method2 
helm upgrade --install skywalking skywalking/skywalking -n skywalking --create-namespace -f skywalking-values.yml
Error: UPGRADE FAILED: cannot patch "skywalking-es-init" with kind Job: Job.batch "skywalking-es-init" is invalid: spec.template: Invalid value:
So I deleted the 'skywalking-es-init' Pod and the corresponding Job, and the upgrade command was successful. However, the logs of the 'skywalking-es-init' Job reported an error.
 [9.4.0-520d531] File:  mysql-service.json layer_entity_name: MYSQL_Service_MySQL-Service conflict with exist configuration id: 2981ef0e-1552-4cf0-a68c-da023b4a0fd8
java.lang.IllegalArgumentException: File:  mysql-service.json layer_entity_name: MYSQL_Service_MySQL-Service conflict with exist configuration id: 2981ef0e-1552-4cf0-a68c-da023b4a0fd8 
Could anyone please help me understand what went wrong and what I can do to fix it?" THX！

**Comment by wu-sheng** [comment]
> [9.4.0-520d531] File: mysql-service.json layer_entity_name: MYSQL_Service_MySQL-Service conflict with exist configuration id: 2981ef0e-1552-4cf0-a68c-da023b4a0fd8
java.lang.IllegalArgumentException: File: mysql-service.json layer_entity_name: MYSQL_Service_MySQL-Service conflict with exist configuration id: 2981ef0e-1552-4cf0-a68c-da023b4a0fd8

This means you are trying to initialize the UI dashboard settings in the ES through init-job, but there is previous versions setup in there. 
`init-job` should only run when the ElasticSearch is wiped clean.

**Reply by maristeslk** [reply]
Could you tell me the correct upgrade steps for SkyWalking, please?

**Reply by wu-sheng** [reply]
The upstream community is always recommending you prepare a new env. New OAP and Elastic deployments, and move the agents targeting to the new deployment. At last, destroy the previous deployment.

Every time, when you face a breaking upgrade. You could follow the above ways. If you insist the smoothly upgrade, that is on your own understanding of SkyWalking kernel and what/why break. 

Ref a previous FAQ, https://skywalking.apache.org/docs/main/next/en/faq/v9-version-upgrade/
